pkg/motor/bucket_resolver.go

```diff
@@ -0,0 +1,118 @@
+package motor
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sonr-io/sonr/internal/bucket"
+	"github.com/sonr-io/sonr/pkg/motor/types"
+	bt "github.com/sonr-io/sonr/x/bucket/types"
+)
+
+func (mtr *motorNodeImpl) NewBucketResolver(context context.Context, did string) (bucket.Bucket, error) {
```

Comment:
In what circumstance would you call this vs `GetBucket`?

Reply:
this is used internally as a helper method. It could be moved to the bucket internal package instead.

Reply:
I think it should then or be made private